File: ejercicio 2/ejercicio 5.js
import java.util.ArrayList;
import java.util.List;
public class ServidorMinecraftCompleto {
    static class Jugador {
        private String nombre;
        private int diamantes;
        public Jugador(String nombre, int diamantes) {
            this.nombre = nombre;
            this.diamantes = diamantes;
        }
        public String getNombre() {
            return nombre;
        }
        public int getDiamantes() {
            return diamantes;
        }
        public int stacks() {
            return diamantes / 64;
        }
        public int diamantesRestantes() {
            return diamantes % 64;
        }
        @Override
        public String toString() {
            return String.format("%s: %d diamantes (%d stacks + %d)", 
                                nombre, diamantes, stacks(), diamantesRestantes());
        }
    }
    private List<Jugador> jugadores;
    private static final int MAX_JUGADORES = 10;
    public ServidorMinecraftCompleto() {
        this.jugadores = new ArrayList<>();
    }
    public void agregar(String nombre, int diamantes) {
        if (jugadores.size() < MAX_JUGADORES) {
            for (Jugador j : jugadores) {
                if (j.getNombre().equalsIgnoreCase(nombre)) {
                    System.out.println("✗ El jugador " + nombre + " ya existe");
                    return;
                }
            }
            jugadores.add(new Jugador(nombre, diamantes));
            System.out.println("✓ " + nombre + " agregado con " + diamantes + " diamantes");
        } else {
            System.out.println("✗ Servidor lleno");
        }
    }
    public void verificarStacks() {
        System.out.println("\n--- STACKS DE DIAMANTES ---");
        for (Jugador j : jugadores) {
            System.out.println("► " + j.getNombre() + ": " + j.stacks() + " stacks");
        }
    }
    public void maxDiamantes() {
        if (!jugadores.isEmpty()) {
            Jugador maxJ = jugadores.get(0);
            for (Jugador j : jugadores) {
                if (j.getDiamantes() > maxJ.getDiamantes()) {
                    maxJ = j;
                }
            }
            System.out.println("\n🏆 Más diamantes: " + maxJ.getNombre() + 
                             " (" + maxJ.getDiamantes() + ")");
        }
    }
    public void totalDiamantes() {
        int total = 0;
        for (Jugador j : jugadores) {
            total += j.getDiamantes();
        }
        System.out.println("💰 Total diamantes: " + total);
        System.out.println("   Equivalente a: " + (total/64) + " stacks + " + (total%64) + " diamantes");
    }
    public static void main(String[] args) {
        System.out.println("=== SERVIDOR MINECRAFT ===\n");
        ServidorMinecraftCompleto server = new ServidorMinecraftCompleto();
        server.agregar("Steve", 128);
        server.agregar("Alex", 65);
        server.agregar("Creeper", 192);
        server.verificarStacks();
        System.out.println("\n--- ESTADÍSTICAS ---");
        server.maxDiamantes();
        server.totalDiamantes();
    }
}

File: ejercicio 2/ejercicio 5.py
class Jugador:
    def __init__(self, nombre, diamantes=0):
        self.nombre = nombre
        self.diamantes = diamantes
    def stacks(self):
        return self.diamantes // 64
    def __str__(self):
        return f"{self.nombre}: {self.diamantes} diamantes ({self.stacks()} stacks)"
class ServidorMinecraft:
    def __init__(self):
        self.jugadores = []
        self.max_jugadores = 10
    def agregar(self, nombre, diamantes):
        if len(self.jugadores) < self.max_jugadores:
            self.jugadores.append(Jugador(nombre, diamantes))
            print(f"✓ {nombre} agregado")
        else:
            print("✗ Servidor lleno")
    def verificar_stacks(self):
        for j in self.jugadores:
            print(f"{j.nombre}: {j.stacks()} stacks")
    def max_diamantes(self):
        if self.jugadores:
            max_j = max(self.jugadores, key=lambda j: j.diamantes)
            print(f"Más diamantes: {max_j.nombre} ({max_j.diamantes})")
    
    def total_diamantes(self):
        total = sum(j.diamantes for j in self.jugadores)
        print(f"Total diamantes: {total}")
        return total
server = ServidorMinecraft()
server.agregar("Steve", 128)
server.agregar("Alex", 65)
server.agregar("Creeper", 192)

print("\n--- Stacks ---")
server.verificar_stacks()

print("\n--- Estadísticas ---")
server.max_diamantes()
server.total_diamantes()